source/grid/grid_in.cc

```diff
@@ -636,6 +636,82 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
                   this->cell_data[section_name] = std::move(temp_data);
                 }
             }
+
+          // Addition of POINT_DATA
+          else if (keyword == "POINT_DATA")
+            {
+              unsigned int n_ids;
+              in >> n_ids;
+
+              AssertThrow(n_ids == vertices.size(),
+                          ExcMessage(
+                            "The VTK reader found a POINT_DATA statement "
+                            "that lists a total of " +
+                            Utilities::int_to_string(n_ids) +
+                            " cell data objects, but this needs to "
+                            "equal the number of vertices (which is " +
+                            Utilities::int_to_string(vertices.size()) + ")"));
+
+
+              in >> keyword;
+
+              // Ignore everything until we get to a SCALARS data set
+              if (keyword == "SCALARS")
+                {
+                  //
+                  std::string field_name;
+                  std::string data_type;
+                  in >> field_name;
+                  in >> data_type;
+
+                  in >> keyword;
+                  AssertThrow(
+                    keyword == "LOOKUP_TABLE",
+                    ExcMessage(
+                      "While reading VTK file, missing keyword 'LOOKUP_TABLE'."));
+
+                  in >> keyword;
+                  AssertThrow(
+                    keyword == "default",
+                    ExcMessage(
+                      "While reading VTK file, missing keyword 'default'."));
+
+                  Vector<double> temp_data;
+                  temp_data.reinit(n_ids);
+                  double data;
+                  for (unsigned int j = 0; j < n_ids; ++j)
+                    {
+                      in >> data;
+                      temp_data[j] = data;
+                    }
+                  this->point_data[field_name] = std::move(temp_data);
+                }
+
+              in >> keyword;
+              // Vector case
+              if (keyword == "VECTORS")
+                {
+                  std::string field_name;
+                  std::string data_type;
+                  in >> field_name;
+                  in >> data_type;
+
+                  Vector<double> temp_data;
+                  temp_data.reinit(n_ids *
+                                   3); // assuming always 3 components per point
```

Comment:
Is there a way to remove this assumption?
```suggestion
                  Vector<double> temp_data(n_ids * 3);
```

Reply:
So far I have always seen three components for point data also in case of 2D fields (the z-component was set to 0). 
Looking at https://docs.vtk.org/en/latest/design_documents/VTKFileFormats.html#dataset-attribute-format, it looks like `VECTORS` is supposed to have three components. 

The safest option I see at the moment is to just throw if we don't have three components per point.

Reply:
I believe VTK always assumes that vectors (and vertex coordinates) are 3d. But if you can write an assertion that checks that, then much the better!